alpha/apps/kaltura/lib/clipconcat/kClipManager.php

```diff
@@ -0,0 +1,509 @@
+		$tempEntry->setConversionProfileId(myPartnerUtils::getConversionProfile2ForPartner($partnerId)->getId());
+		$tempEntry->save();
+		return $tempEntry;
+	}
+
+	/**
+	 * @param string $entryId
+	 * @param  FileSyncKey $concatSyncKey
+	 * @throws Exception
+	 * @throws KalturaAPIException
+	 */
+	private function addDestinationEntryAsset($entryId, $concatSyncKey)
+	{
+		$dbAsset = assetPeer::retrieveOriginalByEntryId($entryId);
+		$dbEntry = entryPeer::retrieveByPK($entryId);
+		$isNewAsset = false;
+		if(!$dbAsset)
+		{
+			$isNewAsset = true;
+			$dbAsset = kFlowHelper::createOriginalFlavorAsset($dbEntry->getPartnerId(), $entryId);
+		}
+
+		if(!$dbAsset)
+		{
+			KalturaLog::err("Flavor asset not created for entry [" . $entryId . "]");
+
+			if($dbEntry->getStatus() == entryStatus::NO_CONTENT)
+			{
+				$dbEntry->setStatus(entryStatus::ERROR_CONVERTING);
+				$dbEntry->save();
+			}
+
+			throw new KalturaAPIException(KalturaErrors::ORIGINAL_FLAVOR_ASSET_NOT_CREATED);
+		}
+
+		$newSyncKey = $dbAsset->getSyncKey(flavorAsset::FILE_SYNC_ASSET_SUB_TYPE_ASSET);
+		kFileSyncUtils::createSyncFileLinkForKey($newSyncKey, $concatSyncKey);
+
+		if($isNewAsset)
+			kEventsManager::raiseEvent(new kObjectAddedEvent($dbAsset));
+		kEventsManager::raiseEvent(new kObjectDataChangedEvent($dbAsset));
+
+		$mediaInfo = mediaInfoPeer::retrieveByFlavorAssetId($dbAsset->getId());
+		if($mediaInfo)
+		{
+			$newMediaInfo = $mediaInfo->copy();
+			$newMediaInfo->setFlavorAssetId($dbAsset->getId());
+			$newMediaInfo->save();
+		}
+
+		if ($dbAsset->getStatus() == asset::ASSET_STATUS_READY)
+		{
+			$dbEntry->syncFlavorParamsIds();
+			$dbEntry->save();
+		}
+	}
+
+	/**
+	 * @param BatchJob $batchJob
+	 * @throws Exception
+	 * @throws KalturaAPIException
+	 */
+	private function concatDone(BatchJob $batchJob)
+	{
+		/** @var kConcatJobData $concatJobData */
+		$concatJobData = $batchJob->getParentJob()->getData();
+		$concatAsset = assetPeer::retrieveById($concatJobData->getFlavorAssetId());
+		$concatSyncKey = $concatAsset->getSyncKey(flavorAsset::FILE_SYNC_ASSET_SUB_TYPE_ASSET);
+		/** @var kClipConcatJobData $clipConcatJobData */
+		$clipConcatJobData = $batchJob->getRootJob()->getData();
+		$this->addDestinationEntryAsset($clipConcatJobData->getDestEntryId(), $concatSyncKey);
+		$this->deleteEntry($clipConcatJobData->getTempEntryId());
+		kJobsManager::updateBatchJob($batchJob->getRootJob(), BatchJob::BATCHJOB_STATUS_FINISHED);
+	}
+
+	/**
+	 * @param $entryId
+	 * @param null $entryType
+	 * @throws Exception
+	 * @throws KalturaAPIException
+	 */
+	private function deleteEntry($entryId, $entryType = null)
+	{
+		$entryToDelete = entryPeer::retrieveByPK($entryId);
+
+		if (!$entryToDelete || ($entryType !== null && $entryToDelete->getType() != $entryType))
+			throw new KalturaAPIException(KalturaErrors::ENTRY_ID_NOT_FOUND, $entryId);
+
+		myEntryUtils::deleteEntry($entryToDelete);
+
+		try
+		{
+			$wrapper = objectWrapperBase::getWrapperClass($entryToDelete);
```

Comment:
Why was this required , since you are deleting temp entry that is not even displayed in search?